### PR TITLE
Correlation prefilters

### PR DESCRIPTION
**Pull request by d-callan**
hello prefilters!

### CODE REVIEW

File: schema/url/computes/correlation.raml [hunk 1]
```diff
@@ -2,9 +2,21 @@
...
+        type: number # should be between 0 and 1 inclusive
+      variance:
+        type: number # should be 0 or greater
+      standardDeviation:
+        type: number # should be 0 or greater
+
   CorrelationComputeConfig:
     properties:
       correlationMethod: CorrelationMethod
+      prefilterThresholds: FeaturePrefilterThresholds
```
Comment (asizemore) [inline schema/url/computes/correlation.raml]: should be optional? We'd have basic defaults i'd think?

File: src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java [hunk 2]
```diff
@@ -85,6 +86,19 @@ protected void execute() {
     String method = computeConfig.getCorrelationMethod().getValue();
     CollectionSpec collectionVariable = computeConfig.getCollectionVariable();
     String entityId = collectionVariable.getEntityId();
+    FeaturePrefilterThresholds featureFilterThresholds = computeConfig.getPrefilterThresholds();
+    String proportionNonZeroThresholdRParam = 
+      featureFilterThresholds != null &&
+      featureFilterThresholds.getProportionNonZero() != null ? 
+        ",proportionNonZeroThreshold=" + featureFilterThresholds.getProportionNonZero() : "";
```
Comment (asizemore) [inline src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java:93]: why not set some defaults? Even if the default is 0. Or did that get added to the microbiomeComputations side?

**Review by asizemore — APPROVED** [review]

File: schema/url/computes/correlation.raml (hunk 1 above)
Comment (d-callan) [reply schema/url/computes/correlation.raml]: When I tested they are functionally optional, but I agree the API should be explicit

File: src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java (hunk 2 above)
Comment (d-callan) [reply src/main/java/org/veupathdb/service/eda/compute/plugins/correlationassaymetadata/CorrelationAssayMetadataPlugin.java:93]: Yes they're in the R

**Pull request merged**